common/db/tag_store/dynamo.go

```diff
@@ -39,24 +46,33 @@ func (d *DynamoTagStore) Clear() error {
 }
 
 func (d *DynamoTagStore) Insert(tag *models.Tag) error {
+	// ensure we perform a consistent read after a write
+	defer d.setConsistentRead(true)
+
 	tag.TagID = time.Now().UnixNano()
 	return d.table.Put(tag).Run()
 }
 
 func (d *DynamoTagStore) Delete(tagID int64) error {
+	// ensure we perform a consistent read after a delete
+	defer d.setConsistentRead(true)
 	return d.table.Delete("TagID", tagID).Run()
 }
 
 func (d *DynamoTagStore) SelectAll() (models.Tags, error) {
+	defer d.setConsistentRead(false)
+
 	tags := models.Tags{}
-	if err := d.table.Scan().Consistent(true).All(&tags); err != nil {
+	if err := d.table.Scan().Consistent(d.consistentRead).All(&tags); err != nil {
 		return nil, err
 	}
 
 	return tags, nil
 }
 
 func (d *DynamoTagStore) SelectByQuery(entityType, entityID string) (models.Tags, error) {
+	defer d.setConsistentRead(false)
+
 	scan := d.table.Scan()
```

Comment:
We should use something like [ElastiCache](https://aws.amazon.com/elasticache/) to cache read requests to Dynamo. I agree that using Dynamo in the way we are currently is not scalable.